### PR TITLE
EPMRPP-76472 || New registration extension point, invalid token message fix

### PR DESCRIPTION
**Pull request by tr1ble**
EPMRPP-76472 || New registration extension point, invalid token message fix

### CODE REVIEW

File: app/src/pages/outside/registrationPage/registrationPage.jsx [hunk 1]
```diff
@@ -111,6 +112,12 @@ const TokenErrorSection = ({ tokenProvided }) => (
       <a className={cx('backlink')} href={referenceDictionary.rpLanding}>
         ReportPortal.io
       </a>
+      <br />
+      <FormattedMessage id={'RegistrationPage.or'} defaultMessage={'or '} />
```
Comment (chivekrodis) [inline app/src/pages/outside/registrationPage/registrationPage.jsx:116]: You set default message as <text>+space so you have `or[space]`, but in translation you have only <text> without any spaces(after here and before in 120line). So please recheck how it looking after selecting different lang(not eng)

File: app/src/pages/outside/registrationPage/registrationPageContainer.jsx [hunk 2]
```diff
@@ -90,13 +105,30 @@ export class RegistrationPageContainer extends Component {
...
-    return !uuid || this.state.isLoadingFinished ? (
-      <RegistrationPage
-        tokenProvided={Boolean(uuid)}
-        tokenActive={this.state.isTokenActive}
-        email={this.state.email}
-        onRegistrationSubmit={this.registrationHandler}
-      />
-    ) : null;
+    const extensions = this.props.extensions;
+    const type = this.props.pageProps.type;
```
Comment (chivekrodis) [inline app/src/pages/outside/registrationPage/registrationPageContainer.jsx:109]: better use destructuring `const {extensions, pageProps: {type}} = this.props;`
also pageProps I guess should be call `query` as it is standard name for `?type=something`

File: app/src/pages/outside/registrationPage/registrationPageContainer.jsx [hunk 3]
```diff
@@ -90,13 +105,30 @@ export class RegistrationPageContainer extends Component {
...
+
+    return (
+      !uuid ||
+      (this.state.isLoadingFinished &&
+        (type === TYPE_SIGNUP && extensions && extensions.length !== 0 ? (
```
Comment (chivekrodis) [inline app/src/pages/outside/registrationPage/registrationPageContainer.jsx:114]: `extensions` is `[]` by default here really necessary  check `extensions &&`?

**Review by chivekrodis — CHANGES_REQUESTED** [review]

**Review by chivekrodis — APPROVED** [review]

**Review by Vadim73i — APPROVED** [review]

File: app/src/pages/outside/registrationPage/registrationPage.jsx [hunk 4]
```diff
@@ -111,6 +112,12 @@ const TokenErrorSection = ({ tokenProvided }) => (
...
+      <a className={cx('backlink')} href="#login">
```
Comment (AmsterGet) [inline app/src/pages/outside/registrationPage/registrationPage.jsx:117]: Please use `Link` from `redux-first-router-link` package to manage page traversing via redux.
See line 222 at [this](app/src/pages/outside/loginPage/pageBlocks/loginBlock/loginForm/loginForm.jsx) file f.e.

**Review by AmsterGet — CHANGES_REQUESTED** [review]

**Review by AmsterGet — APPROVED** [review]

**Pull request merged**